pkg/suggest/suggest.go

```diff
@@ -0,0 +1,58 @@
+// Copyright 2023 The Okteto Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package suggest
+
+// errorSuggestion holds the rules and allows generating suggestions for errors.
+type errorSuggestion struct {
+	rules []ruleInterface
+}
+
+// newErrorSuggestion creates a new errorSuggestion instance.
+func newErrorSuggestion() *errorSuggestion {
+	return &errorSuggestion{}
+}
+
+// withRules adds multiple rules to the errorSuggestion.
+func (es *errorSuggestion) withRules(rules []ruleInterface) *errorSuggestion {
+	for _, r := range rules {
+		es.withRule(r)
+	}
+	return es
+}
+
+// withRule adds a new rule to the errorSuggestion.
+func (es *errorSuggestion) withRule(rule ruleInterface) *errorSuggestion {
+	es.rules = append(es.rules, rule)
+	return es
+}
+
+// suggest applies all rules and returns a user-friendly error
+func (es *errorSuggestion) suggest(err error) error {
+	newErr := err
+	for _, rule := range es.rules {
+		newErr = rule.apply(newErr)
+	}
+	return newErr
+}
+
+// UserFriendlyError generates a user-friendly suggestion for the given error.
+func UserFriendlyError(err error) error {
```

Comment:
An idea could be to have `UserFriendlyError` be a struct that implements the error interface instead of doing it in place which works but we lose all the context. This way it could be used as a regular error while also allowing to potentially expose extra functionality through type casting and enable the use of `error.Is`

Reply:
agree on the struct suggestion, minor comment i think the naming should be perhaps "SuggestionError" as we do have the "UserError" that involves a hint, this new type of error involves a suggestion that is being calculated

Reply:
Change done ✅ Thanks @maroshii 🙇 

About renaming, I am not sure about `SuggestionError` because we have `newErrorSuggestion` too 🤔